python/GafferDelightTest/IECoreDelightPreviewTest/RendererTest.py

```diff
@@ -515,6 +517,40 @@ def runSubstitutions( text, attributes ):
 		self.assertEqual( runSubstitutions( "<attr:test:foo> TEST <attr:test:bar>", { "test:bar" : IECore.StringData( "AAA" ) } ), " TEST AAA" )
 		self.assertEqual( runSubstitutions( "<attr:test:foo> TEST <attr:test:bar>", { "test:foo" : IECore.StringData( "AAA" ), "test:bar" : IECore.StringData( "BBB" ) } ), "AAA TEST BBB" )
 
+	def testMessageHandler( self ) :
+
+		RenderType = GafferScene.Private.IECoreScenePreview.Renderer.RenderType
+
+		for renderType, fileName, expected in (
+			( RenderType.Batch, "", 2 ),
+			( RenderType.Interactive, "", 2 ),
+			( RenderType.SceneDescription, self.temporaryDirectory() + "/test.nsi" , 1 )
+		) :
+
+			with IECore.CapturingMessageHandler() as fallbackHandler :
+
+				handler = IECore.CapturingMessageHandler()
+
+				r = GafferScene.Private.IECoreScenePreview.Renderer.create(
+					"3Delight",
+					renderType,
+					fileName = fileName,
+					messageHandler = handler
+				)
+
+				r.option( "invalid", IECore.BoolData( True ) )
+
+				r.render()
+				time.sleep( 1 )
+				del(r)
```

Comment:
As per previous comments.

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/3767/commits/ec8c1e5db51cfc1435a7ddc77661e04c623844a3

Reply:
I see the `del` is still there. I was under the impression that 3delight wasn't a singleton so we shouldn't need it?

Reply:
Cough, actually pushes the commit with the change. https://github.com/GafferHQ/gaffer/pull/3767/commits/09d39a61676c08f084c7901d1ec2a9c74983f952